### PR TITLE
NCL-565, NCL-786

### PR DESCRIPTION
**Pull request by jsenko**

### CODE REVIEW

**Comment by jsenko** [comment]
@alexcreasy hi, can you verify?

**Comment by jsenko** [comment]
retest this please

**Comment by alexcreasy** [comment]
Sure I'll look at this now

**Comment by jsenko** [comment]
retest this please

**Comment by alexcreasy** [comment]
@jsenko just to put on here what I mentioned on IRC: 

Codestyle wise: Can you create new module pnc.common.buildNotifications and move everything from pnc.build-progress-service AND pnc.websockets in to that module? Then we can remove those two modules and have one common module for the remote notifications.

I want to prioritise this ticket ahead of all others for merging as it is both the most complex and satisfies a requirement of 0.5 release.

@vibe13 your thoughts?

File: rest/src/main/java/org/jboss/pnc/rest/restmodel/BuildRecordRest.java [hunk 1]
```diff
@@ -67,6 +67,7 @@ public BuildRecordRest(BuildRecord buildRecord) {
 
     public BuildRecordRest(BuildTask buildTask) {
         this.id = buildTask.getBuildConfiguration().getId();
```
Comment (vibe13) [inline rest/src/main/java/org/jboss/pnc/rest/restmodel/BuildRecordRest.java]: should be removed

File: ui/app/build-progress-service/_build-progress-service.js [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+      var trackedArray = {};
+
+
+      function test(testerArray, data) {
+        var res = true;
+        testerArray.forEach(function (tester) {
+          if (!tester.test(data)) {
+            res = false;
+          }
+        });
```
Comment (vibe13) [inline ui/app/build-progress-service/_build-progress-service.js]: if one false test is enough, I would put a break? 

if (!tester.test(data)) {
  res = false;
  break;
}

File: ui/app/build-progress-service/_build-progress-service.js [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+        return res;
+      }
+
+      function arrayTest(testerArray, dataArray) {
+        var res = true;
+        dataArray.forEach(function (data) {
+          if (!test(testerArray, data)) {
+            res = false;
```
Comment (vibe13) [inline ui/app/build-progress-service/_build-progress-service.js]: same as prev comment

File: ui/app/build-progress-service/_build-progress-service.js (hunk 2 above)
Comment (jsenko) [inline ui/app/build-progress-service/_build-progress-service.js]: break & return does not work with foreach, i'll use for cycle instead

**Comment by jsenko** [comment]
retest this please

**Comment by vibe13** [comment]
@jsenko @alexcreasy +1 to merge

**Pull request merged**